### ISSUE TITLE
How to get all the cucumber scenario steps in before hook

### ISSUE THREAD

**Issue by devdream02** [cucumber/cucumber-jvm#1641]
I want to access all the cucumber scenario steps in @before hook. Is there a way to do this? 

I have tried passing the cucumber scenario object in the before hook method but it only provides the basic info like scenario.getName() , scenario.getId(). What i require is something like getSteps() which give me the List<String> of all the steps of that particular scenario. 

What i am looking for is something like this

```java
        @Before("@dev")
	public void testcase(Scenario scenario){

		for (Step a : scenario.getSteps()) {
			System.out.println("scenario step = " + a.getText());
		}
	}
```

Basically i want the complete scenario information at the beginning of the test execution itself.

Update:-
If i pass the argument of class cucumber.api.TestCase in the before method then i can access the getTestSteps() method but that leads to below exception. 

```java
cucumber.runtime.CucumberException: When a hook declares an argument it must be of type cucumber.api.Scenario. public void com.thermofisher.bid.spa.kingfisher.ui.steps.Hooks.poc(cucumber.api.TestCase)
```

**Comment by mpkorstanje** [comment]
Please use the [support](https://cucumber.io/support) forums for questions and discussions. We use GitHub issues to track bugs and feature requests exclusively.

You may want to look into the [Plugin](https://github.com/cucumber/cucumber-jvm/blob/master/core/src/main/java/cucumber/api/Plugin.java) system.
